### PR TITLE
fix: back to shopping behaviour

### PR DESCRIPTION
**Pull request by csandru-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/pages/cart.vue [hunk 1]
```diff
@@ -12,9 +12,12 @@
...
 definePageMeta({
-  layoutName: 'checkout',
   pageType: 'static',
 });
 
+const router = useRouter();
 const localePath = useLocalePath();
+const backUrl = localePath(String(router.options.history.state.back));
+const backHref =
+  backUrl === localePath(String(router.options.history.state.current)) ? localePath(paths.home) : backUrl;
```
Comment (ksted) [inline apps/web/pages/cart.vue]:
```suggestion
const routerHistoryState = localePath(String(router.options.history.state));
const backUrl = routerHistoryState.back;
const backHref =
  backUrl === routerHistoryState.current ? localePath(paths.home) : backUrl;
```

Comment (csandru-plenty) [reply apps/web/pages/cart.vue]: thats not gonna work because ```localePath()``` always returns a string in format ```locale-lang-en/app-base-route```

Comment (ksted) [reply apps/web/pages/cart.vue]: Ah, true, although disappointing. How do you feel about this construction then?
```suggestion
const historyState = router.options.history.state;
const backUrl = localePath(String(historyState.back));
const currentUrl = localePath(String(historyState.current));
const homeUrl = localePath(paths.home);

const backHref = backUrl === currentUrl ? homeUrl : backUrl;
```

Comment (csandru-plenty) [reply apps/web/pages/cart.vue]: same as the original, now a bit easier on the eyes 👍 
pushed

**Review by ksted — APPROVED** [review]

**Pull request merged**